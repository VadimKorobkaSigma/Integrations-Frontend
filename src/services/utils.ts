export const parseQuery = (queryString: string) => {
    const query: { [key: string]: string | null } = {};
    const pairs = (queryString[0] === '?' ? queryString.substr(1) : queryString).split('&');
    pairs.forEach((pairRaw) => {
        const pair = pairRaw.split('=');
        query[decodeURIComponent(pair[0])] = decodeURIComponent(pair[1] || '');
    });
    return query;
};

export const normalizeString = (str: string) =>
    Array.from(str)
        .map((char) => char.replace(/[\+\*\?\{\}\(\)\[\]\.\\\|]/, `\\${char}`))
        .join('');
