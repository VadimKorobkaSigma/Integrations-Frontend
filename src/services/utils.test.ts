import { parseQuery, normalizeString } from './utils';

test('Services -> parseQuery', () => {
    expect(parseQuery('')).toMatchSnapshot();
    expect(parseQuery('?a=2')).toMatchSnapshot();
    expect(parseQuery('?a=2&b=3&c')).toMatchSnapshot();
    expect(parseQuery('a=2&b=3&c')).toMatchSnapshot();
});

test('Services -> normalizeString', () => {
    expect(normalizeString('1abc23-+l')).toMatchSnapshot();
    expect(normalizeString('?a=2')).toMatchSnapshot();
    expect(normalizeString('?a=2&b=3&c')).toMatchSnapshot();
    expect(normalizeString('a=@2&b=3/*!&c')).toMatchSnapshot();
});
