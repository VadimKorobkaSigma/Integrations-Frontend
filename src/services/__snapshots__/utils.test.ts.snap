// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`normalizeString 1`] = `"1abc23-\\\\+l"`;

exports[`normalizeString 2`] = `"\\\\?a=2"`;

exports[`normalizeString 3`] = `"\\\\?a=2&b=3&c"`;

exports[`normalizeString 4`] = `"a=@2&b=3/\\\\*!&c"`;

exports[`parseQuery 1`] = `Object {}`;

exports[`parseQuery 2`] = `
Object {
  "a": "2",
}
`;

exports[`parseQuery 3`] = `
Object {
  "a": "2",
  "b": "3",
  "c": "",
}
`;

exports[`parseQuery 4`] = `
Object {
  "a": "2",
  "b": "3",
  "c": "",
}
`;
